### ISSUE TITLE
Crash upon building with expo run:ios : Cannot remove an observer <EXSplashScreenService> for the key path "rootViewController"

### ISSUE THREAD

**Issue by raphaelmsr** [expo/expo#14982]
### Summary

In order to implement react-native-fast-image, I used the expo run:ios on SDK 42.
After upgrading to SDK 43 from SDK 42 on a managed workflow; I removed the react-native-unimodules and followed the guide on GitHub about migrating https://github.com/expo/fyi/blob/master/expo-modules-migration.md

Whenever I build with expo run:ios, the app crashes on start with the following error
`Cannot remove an observer <EXSplashScreenService 0x6000035d23e0> for the key path "rootViewController" from <UIWindow 0x7fd722f13750>
because it is not registered as an observer.`

This bug does not appear when using in Release mode when using the configuration Release flag.
Also expo start in the expo client works correctly as well.

I saw multiple issues regarding expo-splash-screen and I am not sure whether they are connected to the same main issue.
I followed the guides for upgrading and all the documentation I could find; even deleted iOS and android directories to start expo run on a fresh start for Expo SDK 43.

PS : the question about managed and bare workflow is not a bit ambiguous since expo run does allow ios and android directories while being in managed workflow ? I didn't eject at any moment, even expo diagnostics should be updated, as it outputs Bare workflow

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

iOS

### SDK Version (managed workflow only)

43

### Environment

  Expo CLI 4.12.10 environment info:
    System:
      OS: macOS 12.0.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 12.18.3 - /usr/local/bin/node
      Yarn: 1.22.10 - /usr/local/bin/yarn
      npm: 7.20.3 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    Managers:
      CocoaPods: 1.11.2 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.0.1, iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0
    IDEs:
      Android Studio: 4.1 AI-201.8743.12.41.7042882
      Xcode: 13.1/13A1030d - /usr/bin/xcodebuild
    npmPackages:
      expo: ^43.0.1 => 43.0.1 
      react: 17.0.1 => 17.0.1 
      react-dom: 17.0.1 => 17.0.1 
      react-native: 0.64.2 => 0.64.2 
      react-native-web: 0.17.1 => 0.17.1 
      react-navigation: ^4.4.4 => 4.4.4 
    npmGlobalPackages:
      expo-cli: 4.12.10
    Expo Workflow: bare

### Reproducible demo or steps to reproduce from a blank project

In my environment, initialising a new project from scratch didn't reproduce the issue

**Comment by Kudo** [comment]
@raphaelmsr i think the issue is new to sdk43 and thank you to open a new issue. could you share some minimal reproducible example? at least let us know if there are any other 3rd party libraries you installed. maybe there are some compatibility issues between 3rd party libraries.

**Comment by Jonatthu** [comment]
This is happening to me also after upgrading to expo 43SDK

**Comment by RodolfoGS** [comment]
@Kudo I could reproduce the crash with 3 simple steps.
1. Prevent hide the splash with: `SplashScreen.preventAutoHideAsync`
2. Show an `alert`
3. Hide the splash with: `SplashScreen.hideAsync`

**Note**: The crash only happen if you **Do not close** the alert.

Example `App.js`:
```
import * as SplashScreen from 'expo-splash-screen';
import React from 'react';
import { View, Text } from 'react-native';

// 1. prevent hide splash
SplashScreen.preventAutoHideAsync();

// 2. show alert
alert("This is a test\nDO NOT CLOSE THIS ALERT!\nWait 3 seconds...");

// 3. hide splash after 3 seconds
setTimeout(() => {
  SplashScreen.hideAsync();
}, 3000);

export default function App() {
  return (
    <View
      style={{
        flex: 1,
        alignItems: 'center',
        justifyContent: 'space-around',
      }}>
      <Text>Test app</Text>
    </View>
  );
}
```

Tested on a device iPhone X with iOS 15.1. Not tested on Android. Bare workflow.

```
"expo": "~43.0.0",
"expo-splash-screen": "~0.13.3",
```

`expo diagnostics`
```
  Expo CLI 4.12.10 environment info:
    System:
      OS: macOS 12.0.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 14.18.1 - /usr/local/bin/node
      npm: 6.14.15 - /usr/local/bin/npm
      Watchman: 4.9.0 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.11.2 - /usr/local/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.0.1, iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0
      Android SDK:
        API Levels: 28, 29, 30
        Build Tools: 28.0.3, 29.0.2, 29.0.3, 30.0.3
        System Images: android-30 | Google APIs Intel x86 Atom
    IDEs:
      Android Studio: 2020.3 AI-203.7717.56.2031.7784292
      Xcode: 13.1/13A1030d - /usr/bin/xcodebuild
    npmPackages:
      expo: ~43.0.0 => 43.0.1
      react: 17.0.1 => 17.0.1
      react-dom: 17.0.1 => 17.0.1
      react-native: 0.64.2 => 0.64.2
      react-native-web: 0.17.1 => 0.17.1
    npmGlobalPackages:
      expo-cli: 4.12.10
    Expo Workflow: bare
```

**Comment by Kudo** [comment]
@RodolfoGS awesome! your fantastic minimal reproducible example really saves my life. #15026 should fix the issue. will update here once we merged and published new version.

**Comment by raphaelmsr** [comment]
Hi OP here
Sorry for not having updated the ticket 
I actually found the culprit and thus was only happening in Simulator since it was due to an Alert if No Push Token was available
And the code responsible for the crash was actually present in the Expo Notifications documentation on expo.dev that I copy-pasted  in Main.js
Looked like a race condition between Loadin a View and showing an Alert
It's very likely that upon upgrading people find themselves with the same crash !

**Comment by Kudo** [comment]
`expo-splash-screen@0.13.5` published with #15026 fix. please try to update expo-splash-screen and see if the fix works for you. thank you folks to report this issue.

**Comment by RodolfoGS** [comment]
> `expo-splash-screen@0.13.5` published with #15026 fix. please try to update expo-splash-screen and see if the fix works for you. thank you folks to report this issue.

Tested, works perfect. Thank you so much for your help 🎉
